### PR TITLE
add llvm 9 support files (includes llvm 8)

### PR DESCRIPTION
**Pull request by vtjnash**
Just to see where we're at with the upcoming LLVM 9.0.0 release (currently in rc4). Two patches need to be updated, but seems like tests are passing.
```
$ make -C deps uninstall-llvm
$ make -C deps install-llvm USE_BINARYBUILDER_LLVM=0 LLVM_VER=9.0.0rc4
$ make all
```

### CODE REVIEW

**Comment by vchuravy** [comment]
Will also need the new patches from #32712

File: deps/llvm.mk [hunk 1]
```diff
@@ -472,8 +484,27 @@ $(eval $(call LLVM_PATCH,llvm-8.0-D50167-scev-umin))
...
+$(eval $(call LLVM_PATCH,llvm7-revert-D44485))
 endif # LLVM_VER 8.0
 
+ifeq ($(LLVM_VER_SHORT),9.0)
+$(eval $(call LLVM_PATCH,llvm-D27629-AArch64-large_model_6.0.1))
+$(eval $(call LLVM_PATCH,llvm8-D34078-vectorize-fdiv))
+$(eval $(call LLVM_PATCH,llvm-6.0-NVPTX-addrspaces)) # NVPTX -- warning: this fails check-llvm-codegen-nvptx
+$(eval $(call LLVM_PATCH,llvm-7.0-D44650)) # mingw32 build fix
+$(eval $(call LLVM_PATCH,llvm-6.0-DISABLE_ABI_CHECKS))
+#$(eval $(call LLVM_PATCH,llvm7-D50010-VNCoercion-ni)) # TODO
```
Comment (vchuravy) [inline deps/llvm.mk]: Would be good to forward port this patch

**Review by vchuravy — COMMENTED** [review]
Looks like a812a778ef6c05610ddcff33888f37c51bd36a94 snuck in?

File: deps/llvm.mk (hunk 1 above)
Comment (vtjnash) [reply deps/llvm.mk]: Yes, these TODO items need to be taken care of before we could switch to the new version. The goal of this PR is just to get it building everywhere.

**Comment by vtjnash** [comment]
> Looks like a812a77 snuck in?

AFAICT according to the description, that was a 3.9 issue, so it doesn't apply anymore to the updated support files here.

**Comment by staticfloat** [comment]
@vtjnash you were telling me earlier today that some of these patches will need to be included in our BB builder; can you confirm for me that it's just `llvm-8.0-D66401-mingw-reloc.patch`?  It looks like [we already have](https://github.com/staticfloat/LLVMBuilder/tree/master/patches/llvm_patches) the patches for `exegesis` and `test-plugin`.

**Comment by staticfloat** [comment]
Actually, it looks like we already have that one too.

**Comment by vtjnash** [comment]
I think it’s just one (revert-D44485)

**Pull request merged**